arm-visualstudio/2014-04-01-preview/swagger/Csm.json

```diff
@@ -0,0 +1,818 @@
+        }
+      }
+    },
+    "AccountResourceListResult": {
+      "description": "The response to an account resource list GET request.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of resource details.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AccountResource"
+          }
+        }
+      }
+    },
+    "AccountResourceRequest": {
+      "description": "The body of a PUT request to modify a Visual Studio account resource.",
+      "type": "object",
+      "properties": {
+        "accountName": {
+          "description": "The account name.",
+          "type": "string"
+        },
+        "location": {
+          "description": "The Azure instance location.",
+          "type": "string"
+        },
+        "operationType": {
+          "description": "The type of the operation.",
+          "enum": [
+            "unknown",
+            "create",
+            "update",
+            "link"
+          ],
+          "x-ms-enum": {
+            "name": "AccountResourceRequestOperationType"
+          }
+        },
+        "properties": {
+          "description": "The custom properties of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "x-ms-client-flatten": true
+        },
+        "tags": {
+          "description": "The custom tags of the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "CheckNameAvailabilityParameter": {
+      "description": "The body of a POST request to check name availability.",
+      "type": "object",
+      "properties": {
+        "resourceName": {
+          "description": "The name of the resource to check availability for.",
+          "type": "string"
+        },
+        "resourceType": {
+          "description": "The type of resource to check availability for.",
+          "type": "string"
+        }
+      }
+    },
+    "CheckNameAvailabilityResult": {
+      "description": "The response to a name availability request.",
+      "type": "object",
+      "properties": {
+        "message": {
+          "description": "The message describing the detailed reason.",
+          "type": "string"
+        },
+        "nameAvailable": {
+          "description": "The value which indicates whether the provided name is available.",
+          "type": "boolean"
+        }
+      }
+    },
+    "ExtensionResource": {
+      "description": "The response to an extension resource GET request.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AccountResource"
+        }
+      ],
+      "properties": {
+        "plan": {
+          "description": "The extension plan that was purchased.",
+          "$ref": "#/definitions/ExtensionResourcePlan"
+        }
+      }
+    },
```

Comment:
The "ExtensionResource" model inherits from "AccountResource" model. If I understand your service correctly, account resource is the parent of the extension resource. Looking into the example, felt like we are doing it because extension has its own properties "name, type, location, tags, properties", similar properties are there for account as well. If this is true, then would it be better if we inhrerit it from Resource like below, what do you think?

```json
    "ExtensionResource": {
        "description": "The visual studio extension resource",
        "type": "object",
        "allOf": [
          {
            "$ref": "#/definitions/Resource"
          }
        ],
        "properties": {
         "plan": {
              "description": "Extended information about the extension resource plan.",
              "$ref": "#/definitions/ExtensionResourcePlan"
          },
          "properties": {
              "description": "A dictionary of extended properties. This property is currently unused.",
              "type": "object",
              "additionalProperties": {
                 "type": "string"
               },
               "x-ms-client-flatten": true
           },
        }
    }
```

Now, do you think we can use the above same model as response model for both "Extension Resource PUT" and "GET Extension"? (i.e. we get rid of ExtensionResourceRequest model)

This will simplify GET PUT pipeline (if supported), e.g.

```C#
// Get an existing extension resource
//
ExtensionResource extensionResource = client.GetExtension(rgName, accountName, extensionName);
// Add some extended properties
//
extensionResource.properties.Add(key1, value1);
extensionResource.properties.Add(key1, value2);
// Update
//
client.CreateOrUpdate(rgName, accountName, extensionName, extensionResource);
```